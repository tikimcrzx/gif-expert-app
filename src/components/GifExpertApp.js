import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export default function GifExpertApp({ defaultCategories = [] }) {
  // const [categories, setCategories] = useState(['One Punch']);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
}
