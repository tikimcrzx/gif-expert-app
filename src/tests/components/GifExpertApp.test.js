import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../../components/GifExpertApp';

describe('Test at <GifExpertApp />', () => {
  test('Show conten <GifExpertApp />', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('This show list categories', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
