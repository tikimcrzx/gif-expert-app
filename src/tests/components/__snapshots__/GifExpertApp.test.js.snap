// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test at <GifExpertApp /> Show conten <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Test at <GifExpertApp /> This show list categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
